packages/flutter/lib/src/material/button.dart

```diff
@@ -589,3 +589,19 @@ class _RenderInputPadding extends RenderShiftedBox {
     );
   }
 }
+
+/// How the icon in any `button.icon` should be placed where `button` could be:
```

Comment:
Probably a better fit at `button_style_button.dart`, if it is included _only_ for the new buttons.

Reply:
I took a look at the file and it seems that it is a stateful widget and not really related. I think the `button.dart` file is the best place

Reply:
Sorry, but I couldn't follow your point.
The following lines show that _it_ clearly relates to the new buttons:
https://github.com/flutter/flutter/blob/a12e2a473a3214b4556ca589d2d9bd311ac7c6f1/packages/flutter/lib/src/material/button_style_button.dart#L21-L30
The new buttons inherit this class. For instance
https://github.com/flutter/flutter/blob/a12e2a473a3214b4556ca589d2d9bd311ac7c6f1/packages/flutter/lib/src/material/text_button.dart#L57

Reply:
so you suggest I move the enum to `button_style_button.dart` from `button.dart`?

Reply:
CC @Hixie

Reply:
As long as this feature is meant for the new buttons only, I think you should.
However, it may or may not be useful if it is decided to bring this feature to old buttons as well.

Reply:
I found the right file for the `enum`, it is` button_style.dart`. I will be adding it there as a final change for its location where it fits the most. As you said the `button.dart` file is for the obsolete material buttons.